src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,110 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+import { Person } from '../types';
+import { useEffect, useState } from 'react';
+import { PersonLink } from './PersonLink';
+import classNames from 'classnames';
+
+type PeopleTableProps = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<PeopleTableProps> = ({ people }) => {
+  const { slug } = useParams();
+  const [selectPerson, setSelectPerson] = useState<string | null>(null);
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const searchParams = new URLSearchParams(location.search);
+  const query = searchParams.get('query')?.toLowerCase() || '';
+  const centuriesParam = searchParams.get('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order') || 'asc';
+  const sex = searchParams.get('sex');
+
+  useEffect(() => {
+    if (slug) {
+      setSelectPerson(slug);
+    }
+  }, [slug]);
+
+  const findPersonByName = (name: string | undefined | null): Person | null =>
+    name
+      ? people.find(p => p.name.toLowerCase() === name.toLowerCase()) || null
+      : null;
```

Comment:
The `findPersonByName` function is called multiple times for each person in the table, which can be inefficient. Consider optimizing this by creating a map of names to `Person` objects outside of the `map` function to reduce redundant lookups.